cmd/generate/config/rules/generic.go

```diff
@@ -35,7 +37,20 @@ func GenericCredential() *config.Rule {
 		Entropy: 3.5,
 		Allowlists: []config.Allowlist{
 			{
-				StopWords: DefaultStopWords,
+				Description:    "Allowlist for Generic API Keys",
+				MatchCondition: config.AllowlistMatchOr,
+				Regexes: []*regexp.Regexp{
+					regexp.MustCompile(`(?i)` +
+						`(public.?key|public.?token` + // public key -> not a secret
```

Comment:
Personally, I'd suggest grouping common prefixes and suffixes. e.g.,
```suggestion
						`(public[_.-]?(key|token)` + // public key -> not a secret
```

Reply:
> more specific separators

makes sense, especially considering the current regex for including context is [\w.-]{0,10}?, so there will be no other separators.

> grouping

also a great idea